save-demo/src/main/kotlin/com/saveourtool/save/demo/runners/cli/DiktatCliRunner.kt

```diff
@@ -0,0 +1,137 @@
+        val version = if (params.tool == DiktatDemoTool.DIKTAT) {
+            DIKTAT_VERSION
+        } else {
+            KTLINT_VERSION
+        }
+        val fileName = buildString {
+            append(toolName)
+            if (params.tool == DiktatDemoTool.KTLINT) {
+                append("-cli")
+            }
+            if (!osName().startsWith("Linux", ignoreCase = true) && !osName().startsWith("Mac OS", ignoreCase = true)) {
+                append(".cmd")
+            }
+        }
+        val key = ToolKey(toolName, version, fileName)
+        if (toolStorage.doesExist(key).block() != true) {
+            throw FileNotFoundException("Could not find file with key $key")
+        }
+
+        val executableFile = toolStorage.download(key)
+            .writeToFile(workingDir / key.executableName)
+            .block()
+            ?.toFile()
+
+        requireNotNull(executableFile)
+
+        return executableFile
+    }
+
+    override fun getRunCommand(
+        workingDir: Path,
+        testPath: Path,
+        outputPath: Path,
+        configPath: Path?,
+        params: DiktatDemoAdditionalParams
+    ): String = buildString {
+        val executable = getExecutable(workingDir, params)
+        if (osName().startsWith("Linux", ignoreCase = true) || osName().startsWith("Mac OS", ignoreCase = true)) {
+            append("chmod 777 $executable; ")
+        }
+        append(executable)
+        append(" -o $outputPath ")
+        configPath?.let {
+            append(" --config=$it ")
+        }
+        if (params.mode == DiktatDemoMode.FIX) {
+            append(" --format ")
+        }
+        append(testPath)
+    }
+        .also {
+            logger.debug("Running command [$it].")
+        }
+
+    override fun run(
+        testPath: Path,
+        outputPath: Path,
+        configPath: Path?,
+        params: DiktatDemoAdditionalParams,
+    ): DiktatDemoResult {
+        val workingDir = testPath.parent
+        val launchLogPath = workingDir / "log"
+        val command = getRunCommand(workingDir, testPath, outputPath, configPath, params)
+        val processBuilder = createProcessBuilder(command).apply {
+            redirectErrorStream(true)
+            redirectOutput(ProcessBuilder.Redirect.appendTo(launchLogPath.toFile()))
+
+            /*
+             * Inherit JAVA_HOME for the child process.
+             */
+            val javaHome = System.getProperty("java.home")
+            environment()["JAVA_HOME"] = javaHome
+            prependPath(Path(javaHome) / "bin")
+        }
+
+        processBuilder.start().waitFor()
+
+        val logs = launchLogPath.readLines()
+
+        @Suppress("TooGenericExceptionCaught")
+        val warnings = try {
+            outputPath.readLines()
+        } catch (e: Exception) {
+            logs.forEach { log ->
+                logger.error(log)
+            }
+            throw e
+        }
+
+        logs.forEach { log ->
```

Comment:
Up to you, but can also be rewritten as 
```suggestion
        logs.forEach(logger::debug)
```

Reply:
Will use it

Reply:
Done